### PR TITLE
truecrack: init at 3.6

### PR DESCRIPTION
**Pull request by ethancedwards8**
###### Motivation for this change
https://gitlab.com/kalilinux/packages/truecrack

Related to #81418

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/truecrack/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib, gccStdenv, fetchFromGitLab, cudaSupport, cudatoolkit,
+pkg-config }:
```
Comment (jonringer) [inline pkgs/tools/security/truecrack/default.nix:2]: should probably provide the default here
```suggestion
{ lib, gccStdenv, fetchFromGitLab, cudatoolkit
, cudaSupport ? false
, pkg-config }:
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9758,6 +9758,10 @@ in
     inherit (darwin.apple_sdk.frameworks) Security;
   };
 
+  truecrack = callPackage ../tools/security/truecrack { cudaSupport = config.cudaSupport or true; };
+  truecrack-cuda = truecrack.override { cudaSupport = true; };
+  truecrack-free = truecrack.override { cudaSupport = false; };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:9763]:
```suggestion
  truecrack = callPackage ../tools/security/truecrack { };
  truecrack-cuda = truecrack.override { cudaSupport = true; };
```

File: pkgs/tools/security/truecrack/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    pkg-config
+  ];
+
+  buildInputs = lib.optionals cudaSupport [
+    cudatoolkit
+  ];
+
+  cc = "gcc";
+  buildPhase = "make prefix=$out CC=${cc} -j $NIX_BUILD_CORES";
+  installPhase = "make prefix=$out CC=${cc} install";
```
Comment (jonringer) [inline pkgs/tools/security/truecrack/default.nix:32]: CC should already be defined
```suggestion
   installFlags = [ "prefix=$(out)" ];
   enabledParallelBuilding = true;
```

File: pkgs/tools/security/truecrack/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{ lib, gccStdenv, fetchFromGitLab, cudatoolkit
```
Comment (jonringer) [inline pkgs/tools/security/truecrack/default.nix:1]: does it explicitly need gcc?

Comment (ethancedwards8) [reply pkgs/tools/security/truecrack/default.nix:1]: Yes, it fails with clang. And on MacOS Clang is the default stdenv.cc

Comment (jonringer) [reply pkgs/tools/security/truecrack/default.nix:1]: sounds good then

**Comment by ethancedwards8** [comment]
Result of `nixpkgs-review pr 127518` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>truecrack</li>
  </ul>
</details>

**Pull request merged**